### ISSUE TITLE
Ungrouped attribute in cube schema throwing error

### ISSUE THREAD

**Issue by Vigneshg98** [cube-js/cube.js#1093]
**Problem**
I need to query an Elastic index with nested field data, I cannot do group by on it. Hence I am trying to avoid the default GROUP BY in the CubeJS SQL query.

**Request** :
`{"dimensions":["Cubepoc.empid","Cubepoc.empname","Cubepoc.emporg","Cubepoc.emprole","Cubepoc.empexp"],
"measures":[],"timeDimensions":[],"segments":[],"filters":[]}`

**Sample Elastic Index Record** : 
`{
        "_index" : "cubepoc",
        "_type" : "_doc",
        "_id" : "2",
        "_score" : 1.0,
        "_source" : {
          "empid" : "102",
          "empname" : "ragnar lothbrok",
          "emporg" : "google",
          "detail" : [
            {
              "emprole" : "architect",
              "empexp" : "4 years"
            }
          ]
        }
      }`

**Getting the following error while querying** : 
Error: Compile errors:                                                                                                                                   
Cubepoc cube: "ungrouped" is not allowed 

**Related Cube.js schema**
```
cube(`Cubepoc`, {
  sql: `SELECT * FROM cubepoc`,
  measures: {},
  dimensions: {
    empid: {
      sql: `empid`,
      type: `number`,
      title: `Emp Id`,
      primaryKey: true
    },

    empname: {
      sql: `empname`,
      type: `string`,
      title: `Emp Name`
    },

    emporg: {
      sql: `emporg`,
      type: `string`,
      title: `Emp Organisation`
    },

    emprole: {
      sql: `detail.emprole`,
      type: `string`,
      title: `Emp Role`
    },

    empexp: {
      sql: `detail.empexp`,
      type: `string`,
      title: `Emp Experience`
    }
  },
  ungrouped: true
});
```

**Related Cube.js generated SQL**

```sql
SELECT cubepoc.empid cubepoc__empid, cubepoc.empname cubepoc__empname, cubepoc.emporg cubepoc__emporg, detail.emprole cubepoc__emprole, detail.empexp cubepoc__empexp FROM cubepoc AS cubepoc GROUP BY cubepoc.empid, cubepoc.empname, cubepoc.emporg, detail.emprole, detail.empexp ORDER BY cubepoc.empid ASC LIMIT 10000
```

**Comment by vasilev-alex** [comment]
Hey @Vigneshg98! You should pass the `ungrouped` property to the query itself, not define it in the schema. Please refer to https://cube.dev/docs/query-format#query-properties

**Comment by Vigneshg98** [comment]
Hi @vasilev-alex 

Thanks for your response.

I even tried that way. Here's my request `{"dimensions":["Cubepoc.empid","Cubepoc.empname","Cubepoc.emprole"],
"measures":[],"timeDimensions":[],"segments":[],"filters":[],"ungrouped":true}`

Still, the query gets formed including a GROUP BY
`SELECT cubepoc.empid cubepoc__empid, cubepoc.empname cubepoc__empname, detail.emprole cubepoc__emprole FROM cubepoc AS cubepoc GROUP BY cubepoc.empid, cubepoc.empname, detail.emprole ORDER BY cubepoc.empid ASC LIMIT 10000`

And it shows the error like **Grouping isn't (yet) compatible with nested fields [detail.emprole]**  which resembles https://github.com/cube-js/cube.js/issues/1070

Regards !

**Comment by paveltiunov** [comment]
Fixed by #1098
